cpp/src/arrow/compute/api_scalar.h

```diff
@@ -129,6 +129,20 @@ Result<Datum> Multiply(const Datum& left, const Datum& right,
                        ArithmeticOptions options = ArithmeticOptions(),
                        ExecContext* ctx = NULLPTR);
 
+/// \brief Divide two values. Array values must be the same length. If either
+/// factor is null the result will be null.
+///
+/// \param[in] left the dividend
+/// \param[in] right the divisor
+/// \param[in] options arithmetic options (enable/disable overflow checking), optional
+/// \param[in] ctx the function execution context, optional
+/// \return the elementwise quotient, if there is a zero divisor,
+///         an error will be raised
```

Comment:
Can you move the zero divisor mention to the body of the docstring (together with "If either argument is null...")?

Reply:
Done. Thank you for the good suggestion.